### DISCUSSION TITLE
Does this library support SVD?

### DISCUSSION THREAD

**Discussion by joezappie** [josdejong/mathjs#3005] [Ideas]
I'm trying to calculate the transformation between two coordinate systems given 2 sets of data. I'm trying to use Procrustes analysis which calls to use singular value decomposition:

```
Step 4: Perform Singular Value Decomposition (SVD)
Perform SVD on matrix C:
C = U * S * V^T

Where:
U is an orthogonal matrix.
S is a diagonal matrix with singular values in decreasing order.
V^T is the transpose of an orthogonal matrix.
```

Sorry if this is a dumb question - I'm no math expert, never took linear alegbra in college. I'm not sure if SVD is equivalent to something already in the library as there's nothing named SVD directly that I can find. Can I do this with mathjs or will I need another library like ml-matrix or SVD-JS?

**Comment by dvd101x** [comment]
Hi, I don’t think it supports it directly, but you would build your own as it does matrix multiplication, eigenvalues and transpose.

**Comment by gwhitney** [comment]
Right, I don't think mathjs provides a singular value decomposition at the moment. ~~I am about to create a general discussion on mathjs linear algebra capabilities.~~

EDIT: it already exists, #2702.

**Comment by joezappie** [comment]
I did end up using `SVD-JS` for my project, but that project hasnt been updated in 3 years and I'd still love to see SVD in this library so I'll continue to leave this issue open. Unfortunately, my math skills are not good enough to write my own SVD with mathjs's built in matrix multiplication, eigenvalues and transpose functions.

**Reply by josdejong** [reply]
Good to hear. Help implementing SVD in `mathjs` would e welcome!
